src/render/tutorials/api-gateway-security/2_0.html.md

```diff
@@ -0,0 +1,954 @@
+  void givenNoAuthorizationWhenCallBasicAuthApiExpectUnauthorized() {
+    given()
+        .port(8092)
+      .when()
+        .get("/api/secure/basic")
+      .then()
+        .assertThat()
+        .statusCode(401);
+  }
+
+  @Test
+  @DisplayName("GIVEN authorization WHEN call basicAuth API EXPECT Ok")
+  void givenAuthorizationWhenCallBasicAuthApiExpectOk() {
+    given()
+        .port(8092)
+        .header("Authorization", "Basic am9objpzM2NyM3Q=")
+      .when()
+        .get("/api/secure/basic")
+      .then()
+        .assertThat()
+        .statusCode(200);
+  }
+}
+```
+
+We've crated two simple tests using [REST-assured](http://rest-assured.io/). They do the same actions we did above:
+
+- call secured endpoint without authorization and expect `401 - Unauthorized` response
+- call the same endpoint with authorization and expect `200 - OK` response
+
+If you try rebuilding the project now (remember to stop any running containers before that) you'll notice that those test will be invoked. Now we're confident that if our future changes will somehow disable or modify our `/api/secure/basic` logic, we'll know at build time!
+
+## JWT
+
+Now let's do something fancier than just passing the credentials. We'll use [JSON Web Tokens](https://jwt.io/). 
+
+We'll not get into neither cryptography nor JWT specification details, more information is available [here](https://jwt.io/introduction/). 
+
+We'll use [HMAC-SHA256](https://www.ietf.org/rfc/rfc2104.txt) algorithm with a symmetric key. Ideally, we would want to implement a flow as below:
+
+
+![JWT flow](/img/blog/api-gateway-security/security-jwt-flow.png)
+
+1. User logs in with their credentials.
+2. User receives a token encrypted with a key known only to the Server.
+3. User uses the received token to authorize requests to other endpoints. The server uses the key to verify if the token is valid or not.
+
+Since we'll be focusing only on step 3, we'll mimic the result of the other two by generating the token ourselves.
+
+But let's start with design.
+
+### Design & Setup
+
+We'll need a new endpoint and a new security schema:
+
+`knotx/conf/openapi.yaml`
+```yaml
+// ...
+
+paths:  
+  // ...
+  /api/secure/jwt:
+    get:
+      operationId: hello-world-operation-jwt
+      security:
+          - helloJwtAuth: []
+      responses:
+        default:
+          description: Hello World API protected with JWT          
+  // ...
+  
+components:
+  securitySchemes:
+    // ...
+    helloJwtAuth:
+      type: http
+      scheme: bearer
+      bearerFormat: JWT
+```
+
+Again, we defined a new endpoint and attached security schema with JWT bearer auth.
+
+Now, let's take care of our new operation `hello-world-operation-jwt` and security scheme `helloJwtAuth`:
+
+`knotx/conf/routing/operations.conf`
+```hocon
+routingOperations = ${routingOperations} [
+  // ...  
+  {
+    operationId = hello-world-operation-jwt
+    handlers = [
+      {
+        name = hellohandler
```

Comment:
Please note that we reuse the same hello handler, without any changes in the code. Authentication / authorization logic is not a part of the routing handler implementation.